tests/model/test_fully_connected.py

```diff
@@ -115,6 +152,39 @@ def test_fully_connected_raises_value_error_negative_num_hidden_layers(
         FullyConnected(**kwargs)
 
 
+@given(
+    fully_connected_kwargs=fully_connecteds(return_kwargs=True),
+    dropout=st.floats(0.1, 1.0),
+)
+def test_fully_connected_raises_value_error_zero_hidden_dropout_not_None(
+    fully_connected_kwargs: Tuple[FullyConnected, Dict], dropout: float
+) -> None:
+    """Ensures ValueError raised when hidden_size=0 and dropout is not None."""
+    _, kwargs = fully_connected_kwargs
+    kwargs["num_hidden_layers"] = 0
```

Comment:
Yes good point re: separating out  `fully_connected_kwargs` and `fully_connecteds`. I've changed this

Reply:
re using `assume` here, I think your suggested change to `test_fully_connected_raises_value_error_negative_num_hidden_layers` was actually meant for `test_fully_connected_raises_value_error_zero_hidden_dropout_not_None`. 

I'm assuming this because I don't think `fully_connected_kwargs()` should generate a negative number of layers in any case. 

Assuming I'm correct about this but following the spirit of the above comment the two tests now read:
```python
@given(
    kwargs=fully_connected_kwargs(), num_hidden_layers=st.integers(-1000, -1),
)
def test_fully_connected_raises_value_error_negative_num_hidden_layers(
    kwargs: Dict, num_hidden_layers: int
) -> None:
    """Ensures ValueError raised when num_hidden_layers < 0."""
    kwargs["num_hidden_layers"] = num_hidden_layers
    with pytest.raises(ValueError):
        FullyConnected(**kwargs)


@given(
    kwargs=fully_connected_kwargs(), dropout=st.floats(0.0, 1.0),
)
def test_fully_connected_raises_value_error_zero_hidden_dropout_not_None(
    kwargs: Dict, dropout: float
) -> None:
    """Ensures ValueError raised when hidden_size=0 and dropout is not None."""
    assume(kwargs["num_hidden_layers"] == 0)
    kwargs["dropout"] = dropout
    with pytest.raises(ValueError):
        FullyConnected(**kwargs)
```

Reply:
Further to the above, I get a non-deterministic test failure using `assume` for `test_fully_connected_raises_value_error_zero_hidden_dropout_not_None` as follows:
```shell
E   hypothesis.errors.FailedHealthCheck: It looks like your strategy is filtering out a lot of data. Health check found 50 filtered examples but only 9 good ones. This will make your tests much slower, and also will probably distort the data generation quite a lot. You should adapt your strategy to filter less. This can also be caused by a low max_leaves parameter in recursive() calls
E   See https://hypothesis.readthedocs.io/en/latest/healthchecks.html for more information about this. If you want to disable just this health check, add HealthCheck.filter_too_much to the suppress_health_check settings for this test.
```

Thoughts on A) keeping as is and disabling the health-check vs B) returning to:
```python
kwargs["num_hidden_layers"] == 0
kwargs["dropout"] = dropout
```

Reply:
@julianmack: Note that resolution to the above may require changing majority of tests in 'test_fully_connected.py`

Reply:
Okay, assume is too specific here it seems. Now that it's just the `kwargs` that are being generated it seems reasonable to manually set it to `kwargs["num_hidden_layers"] == 0`.